config/src/main/java/com/typesafe/config/impl/Tokenizer.java

```diff
@@ -52,6 +52,14 @@ else if (Character.isISOControl(codepoint))
         return new TokenIterator(origin, input, flavor != ConfigSyntax.JSON);
     }
 
+    static String render(Iterator<Token> tokens) {
+        String renderedText = "";
```

Comment:
StringBuilder would seem more natural for this maybe?

Passing in an iterable rather than an iterator might also avoid some bugs (since iterator is mutable and caller may forget to make a copy)